packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -229,12 +229,14 @@ export class Rule extends Resource implements IRule {
 
         // The actual rule lives in the target stack.
         // Other than the account, it's identical to this one
-        new Rule(targetStack, `${this.node.uniqueId}-${id}`, {
+        const rule = new Rule(targetStack, `${this.node.uniqueId}-${id}`, {
           targets: [target],
-          eventPattern: this.eventPattern,
           schedule: this.scheduleExpression ? Schedule.expression(this.scheduleExpression) : undefined,
           description: this.description,
         });
+        // eventPattern is mutable through addEventPattern(), so we need to lazy evaluate it
```

Comment:
Why not use Lazy.asAny() and avoid hackery?

Reply:
Alternatively expose a proper @internal API for this. Using “as any” should be a very very rare thing if at all.

Reply:
Changed to a subclass and made `@internal`.